File: Java/Tema2_Ejercicio1.java
/*
 * TEMA 2: EXPRESIONES Y OPERADORES - EJERCICIO 1
 * ================================================
 * Escriba un programa que evalúe expresiones matemáticas con operadores 
 * de precedencia. Calcule: (5 + 3) * 2 - 8 / 4 + 10 % 3
 * 
 * EXPLICACIÓN:
 * - Operadores: + (suma), - (resta), * (multiplicación), / (división), % (módulo)
 * - Precedencia: *, /, % se evalúan antes que +, -
 * - Paréntesis tienen la máxima precedencia
 */

public class Tema2_Ejercicio1 {
    public static void main(String[] args) {
        // Cálculo paso a paso
        int resultado1 = (5 + 3);              // = 8
        int resultado2 = resultado1 * 2;       // = 16
        int resultado3 = 8 / 4;                // = 2
        int resultado4 = resultado2 - resultado3;  // = 14
        int resultado5 = 10 % 3;               // = 1
        int resultadoFinal = resultado4 + resultado5;  // = 15
        
        // Forma directa
        int resultado = (5 + 3) * 2 - 8 / 4 + 10 % 3;
        
        System.out.println("=== EVALUACIÓN DE EXPRESIÓN ===");
        System.out.println("Expresión: (5 + 3) * 2 - 8 / 4 + 10 % 3");
        System.out.println("\nPasos:");
        System.out.println("  (5 + 3) = 8");
        System.out.println("  8 * 2 = 16");
        System.out.println("  8 / 4 = 2");
        System.out.println("  16 - 2 = 14");
        System.out.println("  10 % 3 = 1");
        System.out.println("  14 + 1 = 15");
        System.out.println("\nResultado final: " + resultado);
    }
}

/*
 * SALIDA ESPERADA:
 * ================
 * === EVALUACIÓN DE EXPRESIÓN ===
 * Expresión: (5 + 3) * 2 - 8 / 4 + 10 % 3
 * 
 * Pasos:
 *   (5 + 3) = 8
 *   8 * 2 = 16
 *   8 / 4 = 2
 *   16 - 2 = 14
 *   10 % 3 = 1
 *   14 + 1 = 15
 * 
 * Resultado final: 15
 */

File: JS/Tema2_Ejercicio1.js
/*
 * TEMA 2: EXPRESIONES Y OPERADORES - EJERCICIO 1
 * ===============================================
 * Evaluar expresiones respetando precedencia de operadores.
 * 
 * EXPLICACIÓN:
 * - Precedencia de operadores: *, / antes que +, -
 * - Paréntesis para cambiar orden de evaluación
 * - Operadores de comparación y lógicos
 */

// Expresiones matemáticas con precedencia
const resultado1 = 10 + 5 * 2;  // 20 (no 30)
const resultado2 = (10 + 5) * 2;  // 30
const resultado3 = 20 / 4 + 3;  // 8
const resultado4 = 20 / (4 + 2);  // 3.33

// Operadores lógicos
const a = true, b = false;
const andResult = a && b;  // false
const orResult = a || b;   // true
const notResult = !b;      // true

console.log("=== PRECEDENCIA DE OPERADORES ===");
console.log(`10 + 5 * 2 = ${resultado1}`);
console.log(`(10 + 5) * 2 = ${resultado2}`);
console.log(`20 / 4 + 3 = ${resultado3}`);
console.log(`20 / (4 + 2) = ${resultado4}`);

console.log("\n=== OPERADORES LÓGICOS ===");
console.log(`${a} && ${b} = ${andResult}`);
console.log(`${a} || ${b} = ${orResult}`);
console.log(`!${b} = ${notResult}`);

/*
 * SALIDA ESPERADA:
 * === PRECEDENCIA DE OPERADORES ===
 * 10 + 5 * 2 = 20
 * (10 + 5) * 2 = 30
 * 20 / 4 + 3 = 8
 * 20 / (4 + 2) = 3.3333333333333335
 * 
 * === OPERADORES LÓGICOS ===
 * true && false = false
 * true || false = true
 * !false = true
 */
